PyDSS/pyControllers/Controllers/MotorStall.py

```diff
@@ -5,94 +5,186 @@
     pyControllerAbstract.ControllerAbstract` abstract class.
 
-                :param RegulatorObj: A :class:`PyDSS.dssElement.dssElement` object that wraps around an OpenDSS 'Regulator' element
-                :type FaultObj: class:`PyDSS.dssElement.dssElement`
-                :param Settings: A dictionary that defines the settings for the PvController.
-                :type Settings: dict
-                :param dssInstance: An :class:`opendssdirect` instance
-                :type dssInstance: :class:`opendssdirect`
-                :param ElmObjectList: Dictionary of all dssElement, dssBus and dssCircuit objects
-                :type ElmObjectList: dict
-                :param dssSolver: An instance of one of the classed defined in :mod:`PyDSS.SolveMode`.
-                :type dssSolver: :mod:`PyDSS.SolveMode`
-                :raises: AssertionError if 'RegulatorObj' is not a wrapped OpenDSS Regulator element
-
-        """
-
-
+        :param RegulatorObj: A :class:`PyDSS.dssElement.dssElement` object that wraps around an OpenDSS 'Regulator' element
+        :type FaultObj: class:`PyDSS.dssElement.dssElement`
+        :param Settings: A dictionary that defines the settings for the PvController.
+        :type Settings: dict
+        :param dssInstance: An :class:`opendssdirect` instance
+        :type dssInstance: :class:`opendssdirect`
+        :param ElmObjectList: Dictionary of all dssElement, dssBus and dssCircuit objects
+        :type ElmObjectList: dict
+        :param dssSolver: An instance of one of the classed defined in :mod:`PyDSS.SolveMode`.
+        :type dssSolver: :mod:`PyDSS.SolveMode`
+        :raises: AssertionError if 'RegulatorObj' is not a wrapped OpenDSS Regulator element
+
+    """
+        
     def __init__(self, MotorObj, Settings, dssInstance, ElmObjectList, dssSolver):
         super(MotorStall, self).__init__(MotorObj, Settings, dssInstance, ElmObjectList, dssSolver)
         self._class, self._name = MotorObj.GetInfo()
+        assert self._class == "Load", "Motor stall model can only be used with load models"
+        assert MotorObj.NumPhases == 1, "Motor stall model can only be used with single phase loads"
+        
         self.name = "Controller-{}-{}".format(self._class, self._name)
         self._ControlledElm = MotorObj
         self.__Settings = Settings
         self.__dssSolver = dssSolver
-
-        self._ControlledElm.SetParameter('model', 2)
+        self.__dssInstance = dssInstance
+        self.mode = 3
+        self.model_mode = self._ControlledElm.SetParameter('model', self.mode) # 3 - motor, 1 - Standard constant P+jQ
         self._ControlledElm.SetParameter('vminpu', 0.0)
 
         self.kw = self.__Settings['ratedKW']
         S = self.kw / self.__Settings['ratedPF']
         self.kvar = math.sqrt(S**2 - self.kw**2)
-        self._ControlledElm.SetParameter('kw', self.kw)
-        self._ControlledElm.SetParameter('kvar', self.kvar)
+        mode = 2
```

Comment:
This flow seems a bit non-straightforward. `self.mode` gets set to 3, then `mode` is set to 2, and then there is an if/elif branch on `mode`. And then there is `self.model_mode` which gets used in the Update method. Can the code use one mode variable?

Reply:
Yes you can. When a motor stalls, the model type essentially changes, which is what we are tying to capture here